### DISCUSSION TITLE
声码器Hifigan微调

### DISCUSSION THREAD

**Discussion by godspirit00** [PaddlePaddle/PaddleSpeech#2696] [General]
用自己的数据微调了声学模型之后，搭配预训练的声码器合成的语音听起来不太好听，如何进行声码器Hifigan微调？

**Comment by yt605155624** [comment]
HIFIGAN 论文里面的微调，值得是用 A 数据集训练的 HIFIGAN 用于合成 A 的声学模型生成的 mel 的时候可能会有金属感，于是用声学模型生成的 mel 匹配 gt 的 wav 进行微调，并不是用自己的数据微调，建议看下论文
你想获得在自己数据上效果好的 HIFIGAN，建议分两步
1. 用自己的数据 finetune 我们预训练好的 HIFIGAN
2. 如果 1 产出的模型还有金属感，用上述论文的微调方式再微调一下

**Reply by godspirit00** [reply]
感谢大佬，我是参考了NVIDIA NeMo里教程讲的方法，直接拿自己数据训练好的声学模型生成 mel 匹配 gt 的 wav 对已有的hifigan模型进行微调

**Reply by godspirit00** [reply]
我刚刚试听了一下按照我昨天想到的方法finetune出来的模型，输出的音频无法听懂…
所以想试试大佬说的方法
那么求教大佬，第一步`用自己的数据 finetune 我们预训练好的 HIFIGAN`应该怎么做？是否是直接用wav而不用声学模型生成的mel来finetune？但是似乎`finetune.sh`的stage 0就是生成mel
是否是preprocess完后直接走`finetune.sh`的stage 4？

**Reply by yt605155624** [reply]
用自己的数据 finetune 我们预训练好的 HIFIGAN，数据预处理好之后直接走 run.sh 的流程，把我们预训练好的模型放到 exp/default/checkpoints/ 里，新建一个 records.jsonl 指向预训练好的模型，调大 default.yaml 里面的 iter 数，程序会自动加载预训练好的模型继续训练，参考 https://github.com/PaddlePaddle/PaddleSpeech/discussions/1842 这个讨论里面是 finetune 一个 fastspeech2，但是可以参考准备预训练模型的步骤

**Reply by godspirit00** [reply]
按照大佬的步骤跑了300k+的iter，但是合成的音频听起来依然有金属感，甚至个别细节还不如直接用原本的预训练的aishell3的hifigan模型合成的好，求问有可能是为什么？接下来要继续改进的话是该用声学模型产生的mel去finetune对么？
谢谢大佬！

**Reply by yt605155624** [reply]
接下来要继续改进的话是该用声学模型产生的mel去finetune对么，对，这才是缓解金属感的方法

**Reply by MrInouye** [reply]
> 按照大佬的步骤跑了300k+的iter，但是合成的音频听起来依然有金属感，甚至个别细节还不如直接用原本的预训练的aishell3的hifigan模型合成的好，求问有可能是为什么？接下来要继续改进的话是该用声学模型产生的mel去finetune对么？ 谢谢大佬！

我也是想自己微调hifigan，看你说效果还不如预训练的，直接把我劝退了。。。

**Reply by godspirit00** [reply]
> 接下来要继续改进的话是该用声学模型产生的mel去finetune对么，对，这才是缓解金属感的方法

好的，我试试看，谢谢大佬

**Reply by wjddd** [reply]
> > 接下来要继续改进的话是该用声学模型产生的mel去finetune对么，对，这才是缓解金属感的方法
> 
> 好的，我试试看，谢谢大佬

请问您尝试之后效果如何？

**Reply by godspirit00** [reply]
> > > 接下来要继续改进的话是该用声学模型产生的mel去finetune对么，对，这才是缓解金属感的方法
> > 
> > 
> > 好的，我试试看，谢谢大佬
> 
> 请问您尝试之后效果如何？


生成的结果依然是听不懂……但是最近比较忙，所以没得空去再尝试了……

**Reply by godspirit00** [reply]
UPDATE: 在将声学模型和声码器多训练了一些iter之后，效果挺好的，感谢大佬
